src/renderer/base/renderer.cpp

```diff
@@ -623,7 +622,15 @@ bool Renderer::IsGlyphWideByFont(const std::wstring_view glyph)
 // - <none>
 void Renderer::EnablePainting()
 {
-    _pThread->EnablePainting();
+    // When the renderer is constructed, the initial viewport won't be available yet,
+    // but once EnablePainting is called it should be safe to retrieve.
+    _viewport = _pData->GetViewport();

```

Comment:
I don't think it's worth it. Either way it's not going to run, so it's not like it makes any real difference. With a null pointer dereference it'll stop straight away in the debugger, while a throw will silently shut down and you have to look for a log entry to find out what happened. And there's no chance of shipping with a bug like that, so a cleaner shutdown isn't going to benefit the users.

Also, if we really do want to add a null check, this isn't the only place we'd need to put it, because it's possible to be dereferenced earlier than this. To be really "safe", we'd be putting null checks everywhere, and that just feels like pointless bloat.